pynio/service.py

```diff
@@ -53,7 +53,12 @@ def connect(self, blk1, blk2=None):
         # if block exists, add the receiever, otherwise init connection
         receivers = [] if blk2 is None else [blk2.name]
         if connection:
-            connection['receivers'].extend(receivers)
+            cr = connection['receivers']
+            cr.extend(receivers)
+            # remove duplicates but preserve order
+            seen = set()
+            seen_add = seen.add
```

Comment:
What's up with this line? Something I don't know here? Why not just use seen.add(x)